docs/pages/preview/custom-build-config-schema.mdx

```diff
@@ -0,0 +1,280 @@
+
+```yaml
+functions:
+  greetings:
+    name: Say Hi!
+    inputs:
+      - name: name
+        default_value: Hello world
+        # @info #
+        allowed_values: [Hi, Hello, Hey]
+        # @end #
+```
+
+### `outputs`
+
+An output value is expected from a function. For example, a function has an output value of `Hello world`:
+
+```yaml
+functions:
+  greetings:
+    name: Say Hi!
+    # @info #
+    outputs: [value]
+    # @end #
+    command: set-output value Hello
+```
+
+### `command`
+
+Used to define the command to run when a function is executed. It is **required** to define a command for each reusable function. For example, the command `echo "Hello world"` is used to print a message:
+
+```yaml
+functions:
+  greetings:
+    name: Say Hi!
+    # @info #
+    command: echo "Hello world"
+    # @end #
+```
+
+### `shell`
+
+Used to define the default executable shell for a system. For example, the default shell is set to system's shell:
+
+```yaml
+functions:
+  greetings:
+    name: Say Hi!
+    # @info #
+    shell: /bin/sh
+    # @end #
+```
+
+### Built-in EAS functions
+
+EAS provides a built-in reusable function called `upload_artifact` that you can use in a workflow without defining the function definition.
+
+For example, a workflow with the following `steps` will upload an artifact to the EAS servers:
+
+```yaml upload.yml
+build:
+  name: Upload artifacts
+  steps:
+    - eas/upload_artifact:
+        name: Upload application archive
+        inputs:
+          path: fixtures/app-debug.apk
+    - eas/upload_artifact:
+        name: Upload build artifact
+        inputs:
+          type: build-artifact
+          path: assets/icon.png
+```
+
+> **info** Any function that is built-in and provided by EAS must start with the prefix `eas/`.
+
+The `eas/upload_artifact` function has the following properties
+
+| Property      | Description                                                                 |
+| ------------- | --------------------------------------------------------------------------- |
+| `name`        | The name of the step in the reusable function that shows in the build logs. |
+| `inputs`      | Requires input providers (such as `path` or `type`) to be defined.          |
+| `inputs.path` | **Required**. The path to the artifact that is uploaded to the EAS servers. |
+| `inputs.type` | The type of artifact that is uploaded to the EAS servers.                   |
+
+## Use a reusable function in a `build`
+
+For example, a workflow with the following reusable function contains a single command to print a message that is echoed.
+
+```yaml
+functions:
+  greetings:
+    - name: name
+      default_value: Hello world
+    inputs: [value]
+    command: echo "${ inputs.name }, { inputs.value }"
+```
+
+The above function can be used in a `build` as follows:
+
+```yaml
+build:
+  name: Functions Demo
+  steps:
+    # @info #
+    - greetings:
```

Comment:
Maybe it's worth describing that you can override `working_directory`, `name`, and `shell`.

Reply:
Can yuo give me an example how overriding any of those value is possible? Or should we just add that as a note?

Reply:
Let's say that you have a reusable function for listing files in a directory:
```
functions:
  list_files:
    name: List files
    command: ls -la
```
Now, if you use it in your workflow:
```
build:
  name: List files
  steps:
    - list_files
```
It'll list files in the root directory of your project.

You can change this behavior by overriding `working_directory` in the function call:
```
build:
  name: List files
  steps:
    - list_files:
         working_directory: /a/b/c
```